test/noConflictingLifeCycleHooksRule.spec.ts

```diff
@@ -0,0 +1,102 @@
+import { assertFailures, assertSuccess, IExpectedFailure } from './testHelper';
+
+const ruleName = 'no-conflicting-life-cycle-hooks';
+const fails: IExpectedFailure[] = [
+  {
+    endPosition: {
+      line: 4,
+      character: 9
+    },
+    message: 'Implement DoCheck and OnChanges hooks in class Test is not recommended',
+    startPosition: {
+      line: 1,
+      character: 8
+    }
+  }
+];
+
+describe(ruleName, () => {
+  describe('failure', () => {
+    it('should fail when implements both DoCheck and OnChanges hooks', () => {
+      const source = `
+        class Test implements DoCheck, OnChanges {
```

Comment:
TypeScript should fail with a compile-time error as well but I respect the completeness.